crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -124,12 +120,38 @@ impl TimelineBuilder {
         let room = inner.room();
         let client = room.client();
 
-        let timeline_event_handle = room.add_event_handler({
+        let mut room_update_rx = room.subscribe_to_updates();
+        let room_update_join_handle = spawn({
             let inner = inner.clone();
-            move |event, encryption_info: Option<EncryptionInfo>, push_actions: Vec<Action>| {
-                let inner = inner.clone();
-                async move {
-                    inner.handle_live_event(event, encryption_info, push_actions).await;
+            async move {
+                loop {
+                    let update = match room_update_rx.recv().await {
+                        Ok(up) => up,
+                        Err(broadcast::error::RecvError::Closed) => break,
+                        Err(broadcast::error::RecvError::Lagged(_)) => {
+                            warn!("Lagged behind sync responses, resetting timeline");
```

Comment:
A receiver can lag if the system is overloaded, or if the broadcast is overloaded.

If, in this case we reset the timeline, is it something we really want? I believe we have no other choice, but I wonder if clearing the timeline could not generate a “cascade” of actions such as: timeline is reset, then a new sync is needed, but the system is still overloaded and it adds work, so more overload, so a reset etc. Do you think it could happen?

Reply:
Hitting this would mean there were 8 sync responses (with the current limit) that all contained new data for the room, but weren't yet processed. I think if that happens it's basically guaranteed to just be a bug with the processing, rather than things just being slow. And those 8 or 9 requests should take long enough that any user would have lost patience with the app before it gets to this point already 😅